rfcs/0000-tx-valid-since/0000-template.md

```diff
@@ -0,0 +1,45 @@
+---
+Number: "0000"
+Category: Standards Track
+Status: Proposal
+Author: Jinyang Jiang
+Organization: Nervos Foundation
+Created: 2019-03-11
+---
+
+# Transaction valid since
+
+## Abstract
+
+This RFC suggests adding a new consensus rule to prevent a cell to be spent before a certain block timestamp or a block number.
+
+## Summary 
+
+Transaction adds a new `u64` type field `valid_since`, which prevent the transaction to be mined before a certain point, the highest bit of `valid_since` is `type flag`, the other 63 bits represent the actual value `V`, the consensus to validate this field described as follow:
+
+* ignore this validate rule if all bits of `valid_since` is 0.
+* `V` represent a block number when `type flag` is 0, the validation MUST failed if `tip.block_number < V`.
+* `V` represent a block timestamp when `type flag` is 1, the timestamp represented as `V * 512` seconds, the validation MUST failed if `tip.timestamp < V * 512`.
+
+Otherwise, the transaction validate SHOULD continue.
+
+A cell lock script can check the `valid_since` field of a transaction and return invalid when `valid_since` not satisfied condition, to indirectly prevent cell to be spent before a certain block timestamp or a block number.
```

Comment:
Maybe I misunderstand something, why use lock script to check for `valid_since` one more time? My feeling is that the above consensus rule already does the checking, so what's the value added when using lock script?

Reply:
The above consensus rule can prevent tx to be mined before a point, BUT CAN'T prevent an output to be spent before the point. A user who wants to spend an output can always build another tx without valid_since, so check valid_since in lock script is necessary for the time-based fund lock.

Reply:
Understood now, thanks